### ISSUE TITLE
CSV module is slow

### ISSUE THREAD

**Issue by gitlab-importer** [pypy/pypy#1429]
In Heptapod by bitbucket_importer on Mar 21, 2013, 13:41

*Created originally on Bitbucket by [mrjbq7 (John Benediktsson)](https://bitbucket.org/[7.3.7](https://github.com/gitlab-importer/pypy/milestone/7)B8a3aee44-291f-4c49-89fa-b4af51857ba4%7D/)*

**From (bugs.pypy.org)**

 *Attachments:* [csvtest.py](https://pypy-attachments.s3.amazonaws.com/736937129/fc82dfa9234c51f49f9bb2a5a107c05e6ba95854cff7dc33ab96407c156ae9c3/csvtest.py)[csvtest_preload.py](https://pypy-attachments.s3.amazonaws.com/736937129/699e45fdcc6a2f18d5f0355dd33fab41a337bc06c4e3bfe03ee16d3d9376634b/csvtest_preload.py)

**Comment by Neppord** [comment]
I asked Gemini to test if this is still an issue:

> I ran the benchmark (a 1,000,000-row dummy CSV) using the latest versions of CPython and PyPy available on my system. Here are the results:     
                                                                                                                                                  
  --- CPython (v3.14.6) ---                                                                                                                       
                                                                                                                                                  
  • Reading: 0.950 seconds                                                                                                                        
  • Writing: 0.712 seconds                                                                                                                        
  • Total: 1.662 seconds                                                                                                                          
                                                                                                                                                 
  --- PyPy (v2.7.18, PyPy 7.3.23) ---                                                                                                            
                                                                                                                                                  
  • Reading: 1.087 seconds                                                                                                                        
  • Writing: 0.665 seconds                                                                                                                        
  • Total: 1.752 seconds                                                                                                                          
                                                                                                                                                  
 > As you can see, PyPy is actually slightly faster than CPython at writing the CSV, and just a tiny bit slower at reading it (by about ~0.13     ▄
  seconds).                                                                                                                                      █
                                                                                                                                                 █
>  The original bug report from 2013 described PyPy as being multiple times slower. Given these modern benchmark numbers, the performance gap in  █
  the  csv  module has definitely been closed!

I think we can close this issue now

**Comment by mattip** [comment]
Did you run it on windows?

**Comment by Neppord** [comment]
No, I ran it on Linux. I currently do not have access to a windows machine. Do you think a vm would work as a test platform?

**Comment by mattip** [comment]
Maybe? I don't know how deep to dive here, windows is always a second class citizen and we go back and forth about dropping it. I will close this.
